From 744e6cbd1bd3916a1acc883420e264bf74af06b1 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Tue, 10 Mar 2015 14:18:16 +0800
Subject: [PATCH] MLK-10393 video: mxsfb: fix the bus hang issue caused by
 lcdif register access

In 7D platform, the pixel clock should also be enabled
when accesing the lcdif registers. Otherwise, the bus
will be hang.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index 9f9182c..3b1825a 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -662,6 +662,7 @@ static int mxsfb_set_par(struct fb_info *fb_info)
 
 	clk_enable_axi(host);
 	clk_enable_disp_axi(host);
+	clk_enable_pix(host);
 
 	dev_dbg(&host->pdev->dev, "%s\n", __func__);
 	/*
@@ -944,6 +945,7 @@ static int mxsfb_pan_display(struct fb_var_screeninfo *var,
 
 	clk_enable_axi(host);
 	clk_enable_disp_axi(host);
+	clk_enable_pix(host);
 
 	offset = fb_info->fix.line_length * var->yoffset;
 
-- 
1.8.0

